### PR TITLE
filter accepts a Message by default, not an Update

### PR DESCRIPTION
**Pull request by ema-pe**
The method `BaseFilter.filter()` accepts a `Message` by default. Only if
class attribute `update_filter` is set to `True` the method accepts an
`Update`.

This pull request provides a commit that emphasizes this behaviour, because one can misunderstand it.

Closes: #1595

### CODE REVIEW

File: telegram/ext/filters.py [hunk 1]
```diff
@@ -103,15 +103,13 @@ def __repr__(self):
...
         """This method must be overwritten.
 
-        Note:
-            If :attr:`update_filter` is false then the first argument is `message` and of
-            type :class:`telegram.Message`.
-
         Args:
-            update (:class:`telegram.Update`): The update that is tested.
+            message (:class:`telegram.Message`): The message that is tested. If
+            :attr:`update_filter` is True update is `update` of type
```
Comment (Poolitzer) [inline telegram/ext/filters.py]: you mean, message is `update` of type update right? And you missed a comma there ;P

**Comment by Bibo-Joshi** [comment]
@ema-pe I merged master so tests will succeed and changed the last line of the PR description so it will automatically close #1595 on merge :)

**Comment by ema-pe** [comment]
Good!
--
Emanuele Petriglia (ema-pe)

**Comment by Bibo-Joshi** [comment]
After some discussion, we're actually a bit unsure about this change. Hence, I move this PR to Milestone 13. We'll need some more internal discussion before making a decision.

**Comment by Bibo-Joshi** [comment]
We decided to go another way in #2032, so I'll be closing this. Still, thank you for PRing :)

**Pull request closed without merging**